src/coprocessor/codec/mysql/duration.rs

```diff
@@ -244,6 +244,28 @@ impl Duration {
         self.dur = StdDuration::new(self.dur.as_secs(), nanos);
         Ok(self)
     }
+
+    pub fn checked_add(self, rhs: &Duration) -> Option<Duration> {
```

Comment:
Why not return `Result<Duration>`?

Reply:
I think it would be enough to return `Option<Duration>` in `checked_add` and `checked_sub`, and it would be better to return actual error in the builtin function.

Reply:
Ok, how about add annotation for this function:
```
Checked duration addition. Computes self +  rhs, returning None if overflow occurred.
```

Reply:
That sounds good, I will add it.